scripts/dev_setup.sh

```diff
@@ -45,83 +288,66 @@ else
 	exit 1
 fi
 
-cat <<EOF
-Welcome to Libra!
-
-This script will download and install the necessary dependencies needed to
-build Libra Core. This includes:
-	* Rust (and the necessary components, e.g. rust-fmt, clippy)
-	* CMake
-      * Clang
+if [[ $BATCH_MODE == "false" ]]; then
+    welcome_message
+    printf "Proceed with installing necessary dependencies? (y/N) > "
+    read -e input
+    if [[ "$input" != "y"* ]]; then
+	    echo "Exiting..."
+	    exit 0
+    fi
+fi
 
-If you'd prefer to install these dependencies yourself, please exit this script
-now with Ctrl-C.
+PRE_COMMAND=""
+if [ `whoami` != 'root' ]; then
+  PRE_COMMAND="sudo "
+fi
 
-EOF
+mkdir -p ${HOME}/bin
 
-printf "Proceed with installing necessary dependencies? (y/N) > "
-read -e input
-if [[ "$input" != "y"* ]]; then
-	echo "Exiting..."
-	exit 0
+if [[ $"$PACKAGE_MANAGER" == "apt-get" ]]; then
+	[[ $BATCH_MODE == "false" ]] && echo "Updating apt-get......"
+	$PRE_COMMAND apt-get update
 fi
 
-# Install Rust
-echo "Installing Rust......"
-if rustup --version &>/dev/null; then
-	echo "Rust is already installed"
-else
-	curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain stable
-	CARGO_ENV="$HOME/.cargo/env"
-	source "$CARGO_ENV"
-fi
+install_pkg cmake $PACKAGE_MANAGER
+install_pkg clang $PACKAGE_MANAGER
+install_pkg llvm $PACKAGE_MANAGER
+install_pkg curl $PACKAGE_MANAGER
+install_pkg gcc-powerpc-linux-gnu $PACKAGE_MANAGER
 
-# Run update in order to download and install the checked in toolchain
-rustup update
+#need to change....
+install_pkg libssl-dev $PACKAGE_MANAGER  # openssl-devel in centos
+install_pkg pkg-config $PACKAGE_MANAGER  # pkgconfig in centos
```

Comment:
different package names in centos... need to deal with this, and test centos.

Reply:
was planning to do in a different pr

Reply:
built the docker images to test the various opperating systems.  giant pia.  but mostly done (sccache dies when called in alpine).